pyiron_base/utils/error.py

```diff
@@ -145,7 +145,10 @@ def retry(
             return func()
         except error as e:
             logger.warn(
-                f"{msg} Trying again in {delay}s. Tried {i + 1} times so far..."
+                f"{e}"
+            )
+            logger.warn(
+                f"Trying again in {delay}s. Tried {i + 1} times so far..."
```

Comment:
What about 
```suggestion
                f"{msg} - Caused by {e.__class__.__name__}: {e}. Trying again in {delay}s. Tried {i + 1} times so far..."
```
to include the message?

Reply:
I think showing a general `msg` for a catch all errors will be misleading more often than it is right. Therefore, I think `e` should be printed. If you also want `msg` go ahead, I am ok with that.

Reply:
The point could be to inform the user what is currently not working - sometimes the 'normal' error messages are not helpful and it might be good to say something about the funcion which did not work like ' Connecting to database failed: {e.__class__.__name__: {e}. Will retry ....'. I.e. I also would change the message to not be a error message :)

Reply:
The (not explicitly documented) intent for the function is to silence a specific, known exception temporarily. That's why `msg` is printed and not the exception itself, because the assumption is that the calling code expects this exception to happen sometimes.  It sounds like in your case, we catch too broadly, so I think a better fix is to narrow `errors` where ever the error originates in your case.